server/src/main/kotlin/fi/oph/kitu/logging/LogEventService.kt

```diff
@@ -0,0 +1,37 @@
+package fi.oph.kitu.logging
+
+import fi.oph.kitu.ExternalSystem
+import org.slf4j.spi.LoggingEventBuilder
+import org.springframework.stereotype.Service
+import java.net.http.HttpResponse
+
+data class ResponseData(
+    val externalSystem: ExternalSystem,
+    val headers: Map<String, List<String>>,
+    val body: String,
+)
+
+// Singleton instance in order to access the eventDataContext
+@Service
+class LogEventService {
+    var event: LoggingEventBuilder? = null
+    val responses: MutableList<ResponseData> = mutableListOf()
+
+    fun addResponses(vararg pairs: Pair<ExternalSystem, HttpResponse<String>>) =
+        this.responses.addAll(
+            pairs.map { pair ->
+                ResponseData(
+                    externalSystem = pair.first,
+                    headers = pair.second.headers().map(),
+                    body = pair.second.body(),
+                )
+            },
+        )
+
+    fun setResponsesToEvent() = this.event?.addKeyValue("responses", responses)
+
+    fun log(message: String? = null) {
+        setResponsesToEvent()
+        this.event?.log(message)
```

Comment:
Tyhjentääkö joku jossain `this.responses`-listan ennen servicen seuraavaa käyttökertaa?